### PR TITLE
fix: 🐞 randJSON: prevent key overwritten when looping

### PR DESCRIPTION
**Pull request by rodrigofeijao**
- Object key replaced with uuid()
- Reduce replaced with For loop for better performance

✅ Closes: #180

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/falso/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Issue Number: #180

### CODE REVIEW

File: packages/falso/src/lib/rand-json.ts [hunk 1]
```diff
@@ -66,6 +73,7 @@ export function randJSON<Options extends RandomJSONOptions = never>(
   });
 
   if (options?.length) objectSize = options.length;
+  console.debug('=> ', objectSize);
```
Comment (theryansmee) [inline packages/falso/src/lib/rand-json.ts:76]: This needs removing

File: packages/falso/src/lib/rand-json.ts [hunk 2]
```diff
@@ -35,10 +36,16 @@ const generateRandomValue = (): any => {
 };
 
 const generateRandomObject = (length: number) => {
-  return randNumber({ length }).reduce(
-    (o, key) => ({ ...o, [key]: generateRandomValue() }),
-    {}
-  );
+  let o = {};
+  let arr = randNumber({ length });
```
Comment (theryansmee) [inline packages/falso/src/lib/rand-json.ts:40]: This isn't needed anymore

File: packages/falso/src/lib/rand-json.ts [hunk 3]
```diff
@@ -35,10 +36,16 @@ const generateRandomValue = (): any => {
...
+
+  for (let index = 0; index < arr.length; index++) {
```
Comment (theryansmee) [inline packages/falso/src/lib/rand-json.ts:42]:
```suggestion
  for (let index = 0; index < length; index++) {
```

**Review by theryansmee — COMMENTED** [review]
On every other function we use the length option to determine if a single mock, or an array of mocks is returned. I know this function has already been approved previously, but it might be better if to replace `length` with something like `totalKeys`. That way we can use `length` in the same way we do else where.

I've also noticed that we are using `<caption>`s in the comments but this doesn't work correctly in the docs.

**Comment by theryansmee** [comment]
> On every other function we use the length option to determine if a single mock, or an array of mocks is returned. I know this function has already been approved previously, but it might be better if to replace `length` with something like `totalKeys`. That way we can use `length` in the same way we do else where.
> 
> I've also noticed that we are using `<caption>`s in the comments but this doesn't work correctly in the docs.

My suggested changeof `length` would be a breaking change, so what are your thoughts @NetanelBasal & @shaharkazaz ?

**Comment by rodrigofeijao** [comment]
@theryansmee When I was first creating this I though exactly the same about the `length` property, but end up leaving the idea of creating a new property out with afraid of it being quite out of the remaining patter. But I will need to agree that will probably make more sense on keeping `length` as a property that returns an array, and a new one as `keysLength` or `totalKeys` with the amount of keys of the object. 

@NetanelBasal @shaharkazaz let us know what you think

File: packages/falso/src/lib/rand-json.ts [hunk 4]
```diff
@@ -35,10 +36,15 @@ const generateRandomValue = (): any => {
...
 
 const generateRandomObject = (length: number) => {
-  return randNumber({ length }).reduce(
-    (o, key) => ({ ...o, [key]: generateRandomValue() }),
-    {}
-  );
+  let o = {};
+
+  for (let index = 0; index < length; index++) {
+    o = Object.assign(o, {
```
Comment (NetanelBasal) [inline packages/falso/src/lib/rand-json.ts:42]:
```ts
o[randUuid().replace(/-/g, '')] = generateRandomValue()
```

**Comment by theryansmee** [comment]
I'm thinking we should just go with the breaking change. It makes everything a little more alike 😊

**Comment by NetanelBasal** [comment]
@rodrigofeijao any updates here?

**Comment by rodrigofeijao** [comment]
Sorry, have been incredibly busy with some things this week and didn't have the time to look at this, will try to push it during the weekend. Sorry the delay.

**Comment by rodrigofeijao** [comment]
> I'm thinking we should just go with the breaking change. It makes everything a little more alike 😊

@theryansmee as in leaving as it is?

**Comment by theryansmee** [comment]
> > I'm thinking we should just go with the breaking change. It makes everything a little more alike 😊
> 
> 
> 
> @theryansmee as in leaving as it is? 

No, sorry I meant to change it so that length works as it does else where (creating an array of items)

**Comment by rodrigofeijao** [comment]
Will update the code and the tests, and then push it up for you guys to review it

**Comment by theryansmee** [comment]
Awesome, cheers 😊

**Pull request merged**